### PR TITLE
Add a blackbox repeater test

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

**Comment by jparisu** [comment]
This PR requires of https://github.com/eProsima/dev-utils/pull/35 to solve an ASAN problem

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/DDSTestRepeater.cpp [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                11666,
+                11666,
+                ip_version,
+                transport_protocol)
+    );
+    conf->is_repeater = true;
+
+    return conf;
+
+    // return std::make_shared<configuration::InitialPeersParticipantConfiguration>(
```
Comment (irenebm) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/DDSTestRepeater.cpp:100]: remove

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/DDSTestRepeater.cpp [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        configuration::DDSRouterConfiguration ddsrouter_client_configuration_2,
+        configuration::DDSRouterConfiguration ddsrouter_repeater_configuration,
+        uint32_t samples_to_receive = DEFAULT_SAMPLES_TO_RECEIVE,
+        uint32_t time_between_samples = DEFAULT_MILLISECONDS_PUBLISH_LOOP,
+        uint32_t msg_size = DEFAULT_MESSAGE_SIZE)
+{
+    // Check there are no warnings/errors
+    // TODO: Uncomment when having no listening addresses is no longer considered an error by the middleware
+    // TODO: Change threshold to \c Log::Kind::Warning once middleware warnings are solved
+    // test::LogChecker test_log_handler(Log::Kind::Error);
```
Comment (irenebm) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/DDSTestRepeater.cpp:197]: remove?

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (rsanchez15) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/CMakeLists.txt:1]:
```suggestion
# Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/DDSTestRepeater.cpp [hunk 4]
```diff
@@ -0,0 +1,274 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (rsanchez15) [inline ddsrouter_core/test/blackbox/ddsrouter_core/dds/repeater/DDSTestRepeater.cpp:1]:
```suggestion
// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by irenebm — DISMISSED** [review]

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**